src/wallet/rpcwallet.cpp

```diff
@@ -5517,11 +5521,7 @@ UniValue z_mergetoaddress(const UniValue& params, bool fHelp)
                 }
             },
             [](libzcash::SproutPaymentAddress) { },
-            [](libzcash::UnifiedAddress) {
-                throw JSONRPCError(
-                        RPC_INVALID_PARAMETER,
-                        "Invalid parameter, unified addresses are not yet supported.");
-            }
+            [](libzcash::UnifiedAddress) { },
```

Comment:
This changes the answer I gave in the previous PR (https://github.com/zcash/zcash/pull/6569#discussion_r1173121010). Now that UA outputs will be supported, we need `isToSaplingZaddr` to be set correctly here, as well as adding handling for when an Orchard receiver is present and selected.